lib/Transforms/ConvertSecretWhileToStaticFor/ConvertSecretWhileToStaticFor.td

```diff
@@ -0,0 +1,62 @@
+#ifndef LIB_TRANSFORMS_CONVERTSECRETWHILETOSTATICFOR_CONVERTSECRETWHILETOSTATICFOR_TD_
+#define LIB_TRANSFORMS_CONVERTSECRETWHILETOSTATICFOR_CONVERTSECRETWHILETOSTATICFOR_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def ConvertSecretWhileToStaticFor : Pass<"convert-secret-while-to-static-for"> {
+  let summary = "Convert secret scf.while ops to affine.for ops that have constant bounds.";
+  let description = [{
+  Conversion for While-operation that evaluate secret condition to alternative affine For-operation with constant bound(s).
+
+  It replaces the scf.condition operation found in the scf.while loop with an If-operation that takes check the condition, and conditionally executes operations in the while operation's body and yields values.
+  Note: Running this pass alone does not result in a data-oblivious program; we have to run the `--convert-if-to-select` pass to the resulting program to convert the secret-dependent If-operation to a Select-operation.
```

Comment:
Please document the required `max_iter` attribute on the while loop for this pass to succeed. It's in the example, but should be called out more clearly in the text.